tools/Makefile

```diff
@@ -86,6 +86,12 @@ else
 endif
 
 openfst-$(OPENFST_VERSION)/Makefile: openfst-$(OPENFST_VERSION) | check_required_programs
+	# modify src/extensions/far/Makefile.am to make it work with gcc >= 5, while keep its timestamp to avoid autoreconf
+	$(eval make_file=openfst-$(OPENFST_VERSION)/src/extensions/far/Makefile.in)
+	$(eval STAMP="$(shell date -r $(make_file))")
+	sed -i -r 's~am__DEPENDENCIES_1 =$$~& libfstfar.la~' $(make_file)
+	sed -i -E 's~(@LDADD = libfstfarscript.la) \.\.~\1 libfstfar.la ..~' $(make_file)
+	touch -d $(STAMP) $(make_file)
```

Comment:
Modify the makefile of one openfst extension: far, to make it works with gcc >= 5.

This issue is related to the (libtool Inter-library dependencies)\[https://www.gnu.org/software/libtool/manual/libtool.html#Inter_002dlibrary-dependencies\] problem. For shared libraries that depend on another shared lib, this problem would appear.

Reply:
Hello!

I came across a bug at this code block. This section does not work if the locale of the machine is set to another language. The date of the file at line 91 will be outputted in another language, which will not work in the successive touch command. 

```
touch -d "2018Õ╣┤  3µ£ê  3µùÑ Õ£ƒµø£µùÑ 00:08:40 UTC" openfst-1.6.7/src/extensions/far/Makefile.in
touch: invalid date format ÔÇÿ2018Õ╣┤  3µ£ê  3µùÑ Õ£ƒµø£µùÑ 00:08:40 UTCÔÇÖ
Makefile:89: recipe for target 'openfst-1.6.7/Makefile' failed
```

I suggest to set the locale to English at line 91 so the date is outputted correctly:
`$(eval STAMP="$(LANG=en_us_88591; shell date -r $(make_file))")`

Reply:
Yeah, it's a bug. Thank you. Also `LANG=zh_CN.utf8 date` doesn't work for `3.18.6-2.el7.centos.x86_64` in my test. Anyway, it's OK to change to `.. LANG=C date ..`. The change has been done 812596dfe365658d4ba852cd999a4fb032aea3f7.